File: 0189-rotate-array/0189-rotate-array.java
class Solution {
    public void rotate(int[] nums, int k) {
        int len = nums.length;
        if(k == 0 || k == len){
            return;
        }
        k = k%len;
        int[] temp = new int[len];
        int n = 0;
        for(int i=len-k; i<len; i++){
            temp[n++] = nums[i];
        }  
        for(int i=0; i<len-k; i++){
             temp[n++] = nums[i];
        }
        for(int i=0; i<len; i++){
            nums[i] = temp[i];
        }
    }    
}

File: 0189-rotate-array/0189-rotate-array.c
void rotate(int* nums, int numsSize, int k){
        if(k==0 || k==numsSize) return;
        while(k >= numsSize){
            k -= numsSize;
        }
        reverse(nums, 0, numsSize-1);
        reverse(nums, 0, k-1);
        reverse(nums, k, numsSize-1);
    }
    
    void reverse(int* nums, int s, int e){
        while(s < e){
            int temp = nums[s];
            nums[s] = nums[e];
            nums[e] = temp;
            s++;
            e--;
        }
    }
